### ISSUE TITLE
Error setting up platform google_maps / TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by taugusti** [costastf/locationsharinglib#59]
I am trying to make google_maps tracking to work on hass.io (Raspberry pi).
I get the "TypeError: 'NoneType' object is not iterable" error.

I am aware that there are a lot of mentions of this and simular issues. But all is in the closed issues and no solutions are presented. 
So I open this as a new issue.

### Google account
Created new google account and shared tracked user's position to the new user. Logging into maps.google.com with the new user shows the position of the tracked user.
2-factor logon have been disabled.

### configuration.yaml
device_tracker:
  - platform: google_maps
    username: xxx@yyy.zz
    password: xyz

### The log
When hass.io boots I get this in the log:
```
Error setting up platform google_maps
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/homeassistant/components/device_tracker/__init__.py", line 184, in async_setup_platform
    disc_info)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.7/site-packages/homeassistant/components/device_tracker/google_maps.py", line 46, in setup_scanner
    scanner = GoogleMapsScanner(hass, config, see)
  File "/usr/local/lib/python3.7/site-packages/homeassistant/components/device_tracker/google_maps.py", line 66, in __init__
    self._update_info()
  File "/usr/local/lib/python3.7/site-packages/homeassistant/components/device_tracker/google_maps.py", line 78, in _update_info
    for person in self.service.get_all_people():
  File "/usr/local/lib/python3.7/site-packages/locationsharinglib/locationsharinglib.py", line 439, in get_all_people
    people = self.get_shared_people() + [self.get_authenticated_person()]
  File "/usr/local/lib/python3.7/site-packages/locationsharinglib/locationsharinglib.py", line 400, in get_shared_people
    for info in output[0]:
TypeError: 'NoneType' object is not iterable

```

### Possible issues...
When logging into maps.google.com as the new user (on a pc), Google asks me to validate the device by sending me a text message. When I do that I get access... AND - this is despite the fact that 2-factor is disabled...

I am wondering if this could be related to the problem? 

In the google account / security / your devices there is no mention of the Raspberry pi...

I hope someone can help...

**Comment by danieledwardgeorgehitchcock** [comment]
This seems to be a duplicate of the below:

#58 

Please look at comments there - essentially, the library is broken due to Google changing their authentication / cookie structure.

There are some proposed manual patches but, I had no luck in getting them to work.

Looking at it, it requires a whole re-write of the cookie fetch and parsing method which is a tall development task

**Comment by costastf** [comment]
Please have a look at https://github.com/costastf/locationsharinglib/issues/50#issuecomment-448939141 where a manual workaround is specified to create the cookie file. I have the day off today so I will work on this in a while hopefully coming up with something today or tomorrow.

**Comment by taugusti** [comment]
Hi @costastf 
Wow you are fast to close the issue! - It woud be nice if you could leave the issue open until it is fixed...

I really want to try what you suggest - but I do not have the slightest idea where to fiddle around with cookies on the hass.io installation on a Raspberry Pi.

Thanks

**Comment by costastf** [comment]
Hi @taugusti . Apologies for what seemed to you  as an abrupt closing. This is a long (much longer what I would have hoped) standing issue and as stated it will not be fixed in the current code base because it is just too expensive and that is why the issue is closed. I just pushed a branch removing completely the authentication from this library. 
I have started a project https://github.com/costastf/mapscookiegettercli that will interactively drive a browser session so the user can jump through all the hoops and then retrieve the cookies in a form that can be provided to the current library.
Sadly I didn't finish the POC today but I hope to do so tomorrow. I will of course update this thread for everyone that is broken and awaits a fix.
If you want to try the manual fix mentioned above, you don't need to do anything to hass.io. 
In your workstation, interactively log in to google maps and once completely logged in, use the developer tools to look at the cookies and copy paste all relevant values to the provided script. Once the values are filled in and you execute the script, the file created can be hopefully provided as a cookie_file argument to the current library that will hopefully again do what it is supposed to be doing.
I hope that the above clarifies a bit the situation and helps you move forward with your issues, at least until I make a working version of the new tool that will make things much easier to bootstrap this library.
If you have any questions of need any clarification on the above process please don't hesitate to update the thread.

**Comment by taugusti** [comment]
Hi @costastf 
Ok - it sort of makes sense :-)

I just think you should leave one of the issues open with a comment that this will not be fixed so people like me do not think it has been fixed... But ok - sounds great that you are working on a new approach.

Since the library is included inside hass.io, I have decided not to try to make it run as a custom library so I can do all the tricks you propose. So - for now I will go for owntracks even though I like your solution better :-)

Good luck!

/th

**Comment by costastf** [comment]
I have removed the authentication part from locationsharinglib and uploaded version 4 with those changes.
I have made a proof of concept project that drives a browser via selenium and saves the cookies in a compatible with locationsharinglib format. If anyone wants to try out the POC and very crud version because they are feeling adventurous you can follow the documentation at the end of the page https://github.com/costastf/mapscookiegettercli/blob/initial-usability/USAGE.rst .
This is heavily a work in progress and only works for chrome currently, tested only on Mac (but i guess linux should also work) so please do not start creating issues just yet, unless you also want to contribute. I am planing to clean it up and and make it nicer during the upcoming weekends.
I am hoping to have a somewhat properly working version out after the coming weekend supporting chrome for all latest version of Mac, Linux and Windows. 
Of course if anyone is interested in helping out, all help is welcome!

**Comment by costastf** [comment]
Hi all, I have created an alpha of a cookiegetter project that drives a browser through selenium and saves the cookies. Please look at https://github.com/costastf/mapscookiegettercli/blob/master/USAGE.rst for usage (it is quite simple :) ) and report issues at the tracker.

**Comment by shbatm** [comment]
Using the cookiegetter all I get is: locationsharinglib.locationsharinglibexceptions.InvalidCookies: The cookies provided do not provide a valid session.Please authenticate normally and save a valid session again

**Comment by costastf** [comment]
Please read the comment just above my last. On the latest version of the library the cookiegetter cli entrypoint has been removed. Please try to use https://github.com/costastf/mapscookiegettercli/blob/master/USAGE.rst to get your cookies and provide them to the library. cookiegetter command which is part of the previous version of the library is not expected to work and will not be fixed.

**Comment by shbatm** [comment]
@costastf I should clarify: I get the error after following the new instructions and using the `mapscookiegettercli` package linked above.

**Comment by costastf** [comment]
So you successfully interactively log in the service and get a cookie file and you provide that to the library and it fails? Please try the latest version of the library then and get back to me.
